### ISSUE TITLE
Thoughts on the currency representation in the front-end and the currency setting

### ISSUE THREAD

**Issue by isidorosp** [rotki/rotki#845]
As I was working on PR #844, I collected the below notes:

**Consistency of currency representation**
There is room to improve consistency in how currencies are represented throughout the application. 
* Sometimes we have them as "icons" (after #844 this is only through the use of the `cryptocurrency-icons` module in a few elements), other times we have the ticker symbol, and other times the unicode symbol. 
* Currency unicode symbols often overlap (Yen & Yuan; American Dollars, Australian Dollars, Mexican Pesos (yes it uses the $), Canadian Dollars, Taiwanese Dollars, Singaporean Dollars, ...; Swedish Krona & Norwegian Krone, etc.), which can cause user confusion.

_Suggestion_: Define a standardized approach especially considering that we will likely be adding more currencies going forward. 
* In places where a lot of different currencies may be present (e.g. data tables) it makes sense to use currency representations that are unique (in terms of future-proofing, currently this is only `currency.ticker_symbol`).
* In UI elements for selecting currency it should display: full currency name, currency unicode symbol, currency ticket symbol (currently this is not consistent between the currency selector on the home page and the currency selector in the Settings page). 
* We could also make the `currency.unicode_symbol` an amalgam of the country abbreviation and the unicode symbol in cases of overlap (e.g. the USD as the de-facto dollar keeps $, and Mexican Peso symbol could be written as "M$ / MX$ / Mex$, Canadian Dollar C$, CA$, Can$, etc.), thereby allowing for it to be an alternative to `currency.ticker_symbol` when needing to identify currencies unambiguously.

**Currency selection interface and settings**
* I'm not sure how much UI/UX sense it makes to have the currency selector as a) the only setting which is immediately accessible via the toolbar, b) mixed in with other functionality in the top-right area, and c) in the "most important" (rightmost) position of that area which is generally reserved for the User Menu. I'm not saying it's not useful, and it's obviously something users will want to configure ASAP, but is there a better way to go about it (e.g. as part of the onboarding process)?
* _Suggestion_: in ideal to-be state down the road I would imagine that a user sets their regional/language settings (time, language, amount formatting, placement of inline currency symbols before/after an amount etc.), and of course optional overrides.
* _Suggestion_: In general I'm for ease of access in terms of changing settings, but perhaps it can be done in a different manner? What about a "command-palette" like in Sublime/VS Code that would let you navigate to parts of the app and change any setting via your keyboard?

@kelsos  since this is 90% front-end, thoughts?

**Comment by kelsos** [comment]
Most of this stuff was already there, so I didn't touch anything too much. I agree though that we need to figure a way to avoid ambiguity when it comes to currencies, and we need to make sure that we are consistent. 

In my opinion the `ticker_symbol` still makes more sense for table headers so I would suggest keeping it there as it is.

I also agree that I view currency selection as something that the user would not do ofter, however, I have to admit that I am since @LefterisJP was the one that did that I would like his take on it.

Unfortunately, we don't have any information so maybe we could do a survey or something to gather information on how useful it is to the users.

Yes and maybe we could create an easy way for the user to find and set up the first time.

I am not against the command palette part, but I have trouble remembering shortcuts for anything other than the things so I cannot imagine how it would work. If you have any suggestions or a proposal I will be happy to discuss it though. It seems like something that would be nice for power users. Maybe this could extend to navigation in the future too.

**Comment by LefterisJP** [comment]
> I also agree that I view currency selection as something that the user would not do ofter, however, I have to admit that I am since @LefterisJP was the one that did that I would like his take on it.

Yes I also think so. It's simply a remnant from a much simpler app. It can be put into some less prominent position.

The only thing I don't want to do is offer pre-set country options. I can see why it would be pretty convenient. But I am afraid it could put the project in murky legal waters. As I am not an accountant I can't give advice on tax settings so I want the responsibility of settings selection to be on the user.

> Unfortunately, we don't have any information so maybe we could do a survey or something to gather information on how useful it is to the users.

I really don't think we need to do anything like that for the current currency selector. But could be a good idea for other UI questions.

**Comment by isidorosp** [comment]
> > I also agree that I view currency selection as something that the user would not do ofter, however, I have to admit that I am since @LefterisJP was the one that did that I would like his take on it.
> 
> Yes I also think so. It's simply a remnant from a much simpler app. It can be put into some less prominent position.

I will try to come up with some proposals on this together with a re-organization of the settings in the front-end (e.g. I think it's not necessary to differentiate between General and Accounting settings in the User Menu, they can be tabs on one Settings page).

> The only thing I don't want to do is offer pre-set country options. I can see why it would be pretty convenient. But I am afraid it could put the project in murky legal waters. As I am not an accountant I can't give advice on tax settings so I want the responsibility of settings selection to be on the user.

Fully agreed. If eventually we have a "locale/country" option it should only be used to determine presentation settings, namely: language (if we ever support localization), primary currency, number and currency formats, time zone, date format, etc. each of which should be able to be overwritten by the user.

> > Unfortunately, we don't have any information so maybe we could do a survey or something to gather information on how useful it is to the users.
> 
> I really don't think we need to do anything like that for the current currency selector. But could be a good idea for other UI questions.

I'm totally in favor of something this kind of approach, provided we feel like user community involvement is at a high-enough level as to provide substantive feedback, otherwise it's just another thing that takes up additional time.

**Comment by isidorosp** [comment]
Closing this as we also have https://github.com/rotki/rotki/issues/1235 opened which is more specific.

We can pick up locale setting again in a new issue in the future.
